### PR TITLE
chore: [SIW-1242] README.md files enhancement

### PR DESCRIPTION
**Pull request by christianatzeni**
This PR consists in the `README.md` files enhancement. I've updated all the `README.md` file present in the monorepo:
- `README.md` of `io-wallet-support-func` project.
- `README.md` of `io-wallet-user-func` project.
- `README.md` of `io-wallet-common` package..
- `README.md` of `io-wallet` monorepo.

Since `README` files are the first doc that any resource (old or new) reads, I focused on a minimal and essential approach: I have included commands for the usability of the monorepo, without exhaustive and rigorous explanations (there are official documentations for those).

The related task is [SIW-1242](https://pagopa.atlassian.net/browse/SIW-1242).

#### List of Changes

- Updated the `README.md` file of `io-wallet-support-func` (project)
- Updated the `README.md` file of `io-wallet-user-func` (project)
- Updated the `README.md` file of `io-wallet-common` (package)
- Updated the `README.md` file of `io-wallet` (monorepo)

#### Motivation and Context

Make it clear to any new resource the way, purposes and scope of the different projects/packages present in the monorepo. Provide a kind of "tutorial" ready to use, minimal and essential, to facilitate the new resource and allow it to easily use the various services locally.

#### How Has This Been Tested?

Since these are just README.md files, in my opinion the best way to test changes is to experiment directly in the field and see how a new resource "works" by reading those READMEs.

#### Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

[SIW-1242]: https://pagopa.atlassian.net/browse/SIW-1242?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: host.json
Comment (silvicir) [inline host.json:1]: I think this is an oversight

File: README.md [hunk 1]
```diff
@@ -1,113 +1,156 @@
...
-- `Node.js`
-- `yarn`
+This is the `io-wallet` repository, a monorepo that use [NodeJS](https://nodejs.org/), [TypeScript](https://www.typescriptlang.org/), [fp-ts](https://gcanti.github.io/fp-ts/) and different [Azure SDKs](https://azure.github.io/azure-sdk/#javascript) for the IO Wallet app.
 
-The preferred way to set up the local environment is using [nodenv](https://github.com/nodenv/nodenv) to manage `Node.js` installation and `corepack` (included with `Node.js`) to manage the installation of `yarn`.
-Please refer to `.node-version` for the actual runtime version used.
+A monorepo is a single repository containing multiple different projects, with well-defined relationships and responsabilities. The `io-wallet` monorepo contains two main projects under the `app/` folder, present in the root directory of the monorepo:
 
+- `io-wallet-support-func`: it contains all functionalities for assistance and support. It's a simple backend.
+- `io-wallet-user-func`: it contains all functionalities used by the IO Wallet app end users. It's a complex backend.
```
Comment (silvicir) [inline README.md:13]: I would remove the last two sentences, as 'simple' and 'complex' are subjective adjectives

File: README.md [hunk 2]
```diff
@@ -1,113 +1,156 @@
...
 
-## Bundles
-Applications are bundled into zip file into the `./bundles` folder.
+It's strongly recommended to start looking at the code and understanding how Azure functions work from the `io-wallet-support-func` project, even just to have a basic overview. The `io-wallet` monorepo contains a package also, called `io-wallet-common`, present into the `packages/` folder, present in the root directory of the monorepo: this package is used by all two previous projects and it's very important for these. You can see the `io-wallet-common` package like a local NPM package that includes all the common functionalities and utilities of both the `io-wallet-support-func` and `io-wallet-message-func` projects.
```
Comment (silvicir) [inline README.md:14]: I would suggest deleting the first sentence, as it may be subjective

File: README.md [hunk 3]
```diff
@@ -1,113 +1,156 @@
...
+- **Useful Commands**: show a list of useful command, necessary to have a complete overview of the specific project.
+- **Quickstart**: a minimal and essential tutorial to make the project operational in a very short time, without going into a lot of explanations!
 
-## Useful commands
+It is recommended that you read the README.md files of both projects only after reading the current README. In according to this monorepo, in this codepase you can find the `io-wallet.code-workspace` file, a small [JSON](https://www.json.org/) that includes some informations about the location of all projects and packages included in this monorepo. Moreover, in according to the Turborepo documentation, you can find the `turbo.json` file, that includes all necessary configurations about the `turbo` command. Another reference to the existence of multiple node projects is present in the `package.json` file: there is an attribute called `"workspace"`, inside which there are the directories of the two projects (`io-wallet-support-func` and `io-wallet-user-func`) and the package (`io-wallet-common`) present in this monorepo.
```
Comment (silvicir) [inline README.md:25]: The `io-wallet.code-workspace` is a vscode configuration file

File: README.md [hunk 4]
```diff
@@ -1,113 +1,156 @@
...
+- [@azure/cosmos](https://classic.yarnpkg.com/en/package/@azure/cosmos)
+- [tsup](https://classic.yarnpkg.com/en/package/tsup)
+- [vitest](https://classic.yarnpkg.com/en/package/vitest)
 
-```bash
-# to execute COMMAND on WORKSPACE_NAME
-yarn workspace WORKSPACE_NAME run command
-# to execute COMMAD on all workspaces
-yarn workspace foreach run command
+This project uses [Yarn](https://classic.yarnpkg.com/) as depndencies manager and [Turborepo](https://turbo.build/repo/docs) as monorepo manager. Before continuing to read, it is strongly recommended to read both documentations, especially the Turbo one.
```
Comment (silvicir) [inline README.md:39]: there is a typo: "dependencies"

**Review by silvicir — CHANGES_REQUESTED** [review]

**Review by gunzip — CHANGES_REQUESTED** [review]

**Comment by silvicir** [comment]
the file should include instructions on how to start the project so that anyone can get it run even without prior knowledge of it. So you should include all the `az login` steps (as in the Confluence document I showed you)

**Review by silvicir — APPROVED** [review]

**Pull request merged**